accounts-db/src/accounts.rs

```diff
@@ -247,15 +246,16 @@ impl Accounts {
         feature_set: &FeatureSet,
     ) -> Result<Option<NonZeroUsize>> {
         if feature_set.is_active(&feature_set::cap_transaction_accounts_data_size::id()) {
-            let mut compute_budget =
-                ComputeBudget::new(compute_budget::MAX_COMPUTE_UNIT_LIMIT as u64);
-            let _process_transaction_result = compute_budget.process_instructions(
+            let compute_budget_limits = process_compute_budget_instructions(
                 tx.message().program_instructions_iter(),
-                !feature_set.is_active(&remove_deprecated_request_unit_ix::id()),
-                feature_set.is_active(&add_set_tx_loaded_accounts_data_size_instruction::id()),
-            );
+                feature_set,
+            )
+            .unwrap_or_default();
```

Comment:
no change in logic: use default loaded accounts size limit if no valid user input, including failed to process instructions. Feel `unwrap_or_default()` making its intention clearer.